logstash-core/lib/logstash/pipeline.rb

```diff
@@ -323,61 +393,70 @@ def flush_filters(options = {}, &block)
     end
   end
 
+  class FlusherObserver
+    def initialize(logger, flushing)
+      @logger = logger
+      @flushing = flushing
+    end
+
+    def update(time, result, exception)
```

Comment:
Well, this is part of the concurrent ruby package. I don't believe that we should document external libraries here. As to what its used for, it is clearly used in the function immediately following its definition. I don't think we need to add a comment explaining that given that fact.

As we transition toward concurrent ruby's timer task and away from stud buffer this will simply become more familiar IMHO, and thus shouldn't be documented any more than we'd document a specific idiom in stud.